postgres/database.go

```diff
@@ -40,6 +40,9 @@ var (
 	putPgStr    = "INSERT INTO public.blocks (key, data, block_number) VALUES ($1, $2, $3) ON CONFLICT DO NOTHING"
 	deletePgStr = "DELETE FROM public.blocks WHERE key = $1"
 	dbSizePgStr = "SELECT pg_database_size(current_database())"
+
+	getStateLeafKeyPgStr   = "SELECT state_leaf_key FROM eth.state_cids WHERE mh_key = $1"
```

Comment:
Since you could have the same trie node IPLD at different block heights, we run into some potential issues if all we consider when following a link from `public.blocks` back to `state_cids` or `storage_cids` is the `mh_key` and not also `block_number`.